testutil/accumulator.go

```diff
@@ -161,6 +161,17 @@ func (a *Accumulator) Get(measurement string) (*Metric, bool) {
 	return nil, false
 }
 
+// Calls the given Gather function and returns the first error found.
+func (a *Accumulator) GatherError(gf func(telegraf.Accumulator) error) error {
+	if err := gf(a); err != nil {
+		return err
+	}
+	if len(a.Errors) > 0 {
+		return a.Errors[0]
+	}
+	return nil
+}
```

Comment:
Long term I think we need to allow access to log level as well, probably using a good logging library that allows control at a per logger basis (and one logger per plugin).